"should work with two schema models in a hasMany association" test undeterministic failure

## What you are doing?
I am running the `should work with two schema models in a hasMany association` test with postgres or mysql on the master branch.

`GREP='should work with two schema models in a hasMany association' npm run test-integration-postgres`
or 
`GREP='should work with two schema models in a hasMany association' npm run test-integration-mysql`
 

## What do you expect to happen?
I expect the test to pass every time.

## What is actually happening?
Every 5th or 6th time, the test manages to fail, and it can fail in multiple ways

```
  1) [POSTGRES] Include separate should work with two schema models in a hasMany association:

      AssertionError: expected 'a' to equal 'b'
      + expected - actual

      -a
      +b
```
or
```
1) [POSTGRES] Include separate should work with two schema models in a hasMany association:

      AssertionError: expected 'd' to equal 'b'
      + expected - actual

      -d
      +b
```

__Dialect:__ postgres | mysql

--- a/test/integration/include/seperate.test.js
+++ b/test/integration/include/seperate.test.js
@@ -378,6 +378,7 @@ if (current.dialect.supports.groupedLimit) {
       it('should work with two schema models in a hasMany association', function() {
         var User = this.sequelize.define('User', {}, {schema: 'archive'})
           , Task = this.sequelize.define('Task', {
+              id: { type: DataTypes.INTEGER, primaryKey: true },
               title: DataTypes.STRING
             }, {schema: 'archive'});
 
@@ -389,19 +390,19 @@ if (current.dialect.supports.groupedLimit) {
               return Promise.join(
                 User.create({
                   tasks: [
-                    {title: 'b'},
-                    {title: 'd'},
-                    {title: 'c'},
-                    {title: 'a'}
+                    {id: 1, title: 'b'},
+                    {id: 2, title: 'd'},
+                    {id: 3, title: 'c'},
+                    {id: 4, title: 'a'}
                   ]
                 }, {
                   include: [User.Tasks]
                 }),
                 User.create({
                   tasks: [
-                    {title: 'a'},
-                    {title: 'c'},
-                    {title: 'b'}
+                    {id: 5, title: 'a'},
+                    {id: 6, title: 'c'},
+                    {id: 7, title: 'b'}
                   ]
                 }, {
                   include: [User.Tasks]
@@ -415,6 +416,7 @@ if (current.dialect.supports.groupedLimit) {
                 ],
                 logging: console.log
               }).then((result) => {
+
                 expect(result[0].tasks.length).to.equal(2);
                 expect(result[0].tasks[0].title).to.equal('b');
                 expect(result[0].tasks[1].title).to.equal('d');

--- a/test/integration/include/seperate.test.js
+++ b/test/integration/include/seperate.test.js
@@ -416,7 +416,6 @@ if (current.dialect.supports.groupedLimit) {
                 ],
                 logging: console.log
               }).then((result) => {
-
                 expect(result[0].tasks.length).to.equal(2);
                 expect(result[0].tasks[0].title).to.equal('b');
                 expect(result[0].tasks[1].title).to.equal('d');

